pkg/workload/tpcc/README.md

```diff
@@ -0,0 +1,312 @@
+- [Warehouse Configuration](#warehouse-configuration)
+- [Single Region Setup](#single-region-setup)
+  - [Connection Distribution with Partitioning](#connection-distribution-with-partitioning)
+  - [Connection Calculation](#connection-calculation)
+  - [Worker Calculation](#worker-calculation)
+  - [Client Partition Configuration](#client-partition-configuration)
+  - [Running Workload with Client-side Data Partitioning Example](#running-workload-with-client-side-data-partitioning-example)
+- [Multi-region Setup](#running-tpc-c-in-multi-region-setup)
+  - [Prerequisites](#prerequisites)
+  - [Multi-region Configuration](#multi-region-configuration)
+    - [Initial Data Loading and Warehouse Distribution](#initial-data-loading-and-warehouse-distribution)
+    - [Connection and Worker Configuration](#connection-and-worker-configuration)
+  - [Example Setup](#example-setup)
+- [Performance Tuning](#performance-tuning)
+  - [Max Rate Configuration](#max-rate-configuration)
+
+TPC-C is an industry-standard OLTP benchmark that simulates a complex order-entry system to evaluate database performance.
+
+## Basic Usage
+Import Data:
+```bash
+cockroach workload fixtures import tpcc <PGURL> --warehouses=1000
+```
+Run workload:
+```bash
+cockroach workload run tpcc \
+  --warehouses=100 \
+  --db=tpcc \
+  --duration=1h \
+  --ramp=10m
+```
+### Essential Flags
+```bash
+--warehouses=N          # Total number of warehouses
+--active-warehouses=N   # Number of active warehouses (defaults to --warehouses)
+--workers=N             # Number of concurrent workers. Defaults to --warehouses * 10
+--duration=1h           # The duration to run (in addition to --ramp). If 0, run forever.
+--ramp=10m              # Initial period where workers linearly scale from 1 to --workers. The duration over which to ramp up load.
+```
+
+## Warehouse Configuration
+The TPC-C workload operates on a set of warehouses, where you can specify both total warehouses and active warehouses. 
+Active warehouses are the ones that receive traffic during the workload run.
+
+Consider a configuration with:
+- Total warehouses: 10,000
+- Active warehouses: 5,000
+
+This means:
+- The database will be populated with 10,000 warehouses
+- Only 5,000 warehouses will receive actual traffic during the workload run
+
+
+## Single-region Setup
+When using multiple workload generators within a single-region CockroachDB cluster, workload partitioning is essential. 
+This feature divides warehouses among workload nodes, with each node targeting only its assigned warehouses. Without 
+partitioning, nodes would compete for the same warehouses, causing transaction contention and deadlocks that reduce 
+performance.
+
+```bash
+--client-partitions=N    # Number of partitions to divide warehouses into
+--partition-affinity=K   # Specifies which partition(s) this client targets, from 0 to N-1. Multiple values can be specified using commas
+```
+
+Consider a configuration with:
+- Total warehouses: 10,000
```

Comment:
For all examples, it might improve readability to include the CLI args in parentheses, i.e., `(--warehouses=10000)`, etc.